tensorly/regression/cp_plsr.py

```diff
@@ -220,8 +224,12 @@ def transform(self, X, Y=None):
                 Y_scores = T.index_update(
                     Y_scores, T.index[:, a], T.dot(Y, self.Y_factors[1][:, a])
                 )
+
                 Y -= T.dot(
-                    T.dot(T.dot(X_scores, T.pinv(X_scores)), Y_scores[:, [a]]),
+                    T.dot(
+                        T.lstsq(T.transpose(X_scores), T.transpose(X_scores))[0],
```

Comment:
Hmm, is running least square on X_scores against X_scores right?

Reply:
I agree it doesn't look right. I only replaced it with the expression equivalent to what was there before.

Reply:
Hmm I think the equivalent one should look like `T.dot(X_scores, T.lstsq(X_scores, Y_scores[:, [a]])[0])`

Reply:
I confirmed that my commit gave the exact same results before and after. I agree that your line is more likely right, though.